### ISSUE TITLE
Failed at the mastodon@ start script 'babel-node ./streaming/index.js --presets es2015,stage-2

### ISSUE THREAD

**Issue by simbalion** [tootsuite/mastodon#2669]
mastodon streaming service keeps crashing at start. 

Issue #1045 was similar to this however the npm-debug.log output in my situation is slightly different, and the solutions from #1045 did not resolve this.

Here is the excerpt from /home/mastodon/live/npm-debug.log:

0 info it worked if it ends with ok
1 verbose cli [ '/usr/bin/nodejs', '/usr/bin/npm', 'run', 'start' ]
2 info using npm@2.15.11
3 info using node@v4.8.2
4 verbose run-script [ 'prestart', 'start', 'poststart' ]
5 info prestart mastodon@
6 info start mastodon@
7 verbose unsafe-perm in lifecycle true
8 info mastodon@ Failed to exec start script
9 verbose stack Error: mastodon@ start: babel-node ./streaming/index.js --presets es2015,stage-2
9 verbose stack spawn ENOENT
9 verbose stack at ChildProcess. (/usr/lib/node_modules/npm/lib/utils/spawn.js:17:16)
9 verbose stack at emitTwo (events.js:87:13)
9 verbose stack at ChildProcess.emit (events.js:172:7)
9 verbose stack at maybeClose (internal/child_process.js:862:16)
9 verbose stack at Process.ChildProcess._handle.onexit (internal/child_process.js:222:5)
10 verbose pkgid mastodon@
11 verbose cwd /home/mastodon/live
12 error Linux 3.16.0-4-amd64
13 error argv "/usr/bin/nodejs" "/usr/bin/npm" "run" "start"
14 error node v4.8.2
15 error npm v2.15.11
16 error file sh
17 error code ELIFECYCLE
18 error errno ENOENT
19 error syscall spawn
20 error mastodon@ start: babel-node ./streaming/index.js --presets es2015,stage-2
20 error spawn ENOENT
21 error Failed at the mastodon@ start script 'babel-node ./streaming/index.js --presets es2015,stage-2'.
21 error This is most likely a problem with the mastodon package,
21 error not with npm itself.
21 error Tell the author that this fails on your system:
21 error babel-node ./streaming/index.js --presets es2015,stage-2
21 error You can get information on how to open an issue for this project with:
21 error npm bugs mastodon
21 error Or if that isn't available, you can get their info via:
21 error
21 error npm owner ls mastodon
21 error There is likely additional logging output above.
22 verbose exit [ 1, true ]

* * * *

- [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.
- [ ] This bug happens on a [tagged release](https://github.com/tootsuite/mastodon/releases) and not on `master` (If you're a user, don't worry about this).

**Comment by ashfurrow** [comment]
Hi @simbalion – did you ever get this resolved? Would love to know what solution worked.

**Comment by simbalion** [comment]
I did, I had to install node.js again, something had not gone right the first time. Also the npm install -g yarn step needed to be run as the mastodon user from the live/ folder, I think the instructions said to run it as root. There were a few other issues too, too much to recall or I'd recommend fixes to the instructions. Safe to say tho that the debian instructions probably should be re-visited.

My instance is functioning today at https://pridelands.io :)
